File: examples/use_after_move_var.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    string origin = "This is a string.";

    string moved = std::move(origin);

    cout << "origin after move: \"" << origin << "\"" << endl;
    cout << "moved  after move: \"" << moved  << "\"" << endl;

    // Binding that was moved out from can be reassigned
    origin = "Hello, I'm back!";
    cout << "origin after being reassigned: \"" << origin << "\"" << endl;
}

File: examples/use_after_move_var.rs
fn main() {
    let mut origin: String = "This is a string".into();

    let moved = origin;

    println!("origin after move: {}", origin);
    println!("moved  after move: {}", moved);

    // Binding that was moved out from can be reassigned
    origin = "Hello, I'm back!".into();
    println!("origin after reassignment: {}", origin);
}
